controllers/discovery/searchTopic.js

```diff
@@ -23,7 +26,12 @@ const Search = async (req, res) => {
             "Topic".*,
             count("topicFollower"."user_id") 
                 AS "followersCount",
-            (SELECT "f"."user_id" AS "user_id_follower" FROM "user_topics" AS f WHERE "f"."user_id" = :userId AND "f"."topic_id" = "Topic"."topic_id")
+            (SELECT "f"."user_id" AS "user_id_follower" 
+              FROM "user_topics" AS f 
+              WHERE 
+                ("f"."user_id" = :userId 
+                OR "f"."user_id" = :anonymousUserId)
+                AND "f"."topic_id" = "Topic"."topic_id")
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The SQL query has been updated to include conditions for the anonymous user. Ensure that this change doesn't affect the performance of the query significantly, especially when dealing with large datasets.
```diff
-            (SELECT "f"."user_id" AS "user_id_follower" FROM "user_topics" AS f WHERE "f"."user_id" = :userId AND "f"."topic_id" = "Topic"."topic_id")
+             (SELECT "f"."user_id" AS "user_id_follower" 
+               FROM "user_topics" AS f 
+               WHERE 
+                 ("f"."user_id" = :userId 
+                 OR "f"."user_id" = :anonymousUserId)
+                 AND "f"."topic_id" = "Topic"."topic_id")
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->